client/gutenberg/extensions/videopress/editor.js

```diff
@@ -8,12 +8,36 @@ import { addFilter } from '@wordpress/hooks';
 /**
  * Internal dependencies
  */
-import isJetpackExtensionAvailable from 'gutenberg/extensions/presets/jetpack/utils/is-jetpack-extension-available';
 import withVideoPressEdit from './edit';
+import withVideoPressSave from './save';
+import { isEnabled } from 'config';
 
-const addVideoPressSupport = ( settings, name ) =>
-	'core/video' === name ? { ...settings, edit: withVideoPressEdit( settings.edit ) } : settings;
+const addVideoPressSupport = ( settings, name ) => {
+	if ( 'core/video' !== name ) {
+		return settings;
+	}
 
-if ( isJetpackExtensionAvailable( 'videopress' ) ) {
+	return {
+		...settings,
+		attributes: {
+			...settings.attributes,
+			guid: {
+				type: 'string',
+			},
+		},
+		edit: withVideoPressEdit( settings.edit ),
+		save: withVideoPressSave( settings.save ),
+		deprecated: [
+			{
+				attributes: settings.attributes,
+				save: settings.save,
+			},
+		],
+	};
+};
+
+// TODO: Check availability using the availability endpoint. Since this extension is not registered on the server yet,
+// we need to bypass the `isJetpackExtensionAvailable` check
+if ( isEnabled( 'jetpack/blocks/beta' ) ) {
```

Comment:
For folks reading. We will be able to get rid of this once https://github.com/Automattic/jetpack/pull/11212 lands and we set the videopress extension as available using `jetpack_set_extension_available`.